programs/temp/src/instructions/create_nft.rs

```diff
@@ -11,30 +14,18 @@ use anchor_spl::token::{Mint, Token};
 pub struct MintNFT<'info> {
     #[account(mut)]
     pub mint_authority: Signer<'info>,
-    /// CHECK: This is not dangerous because we don't read or write from this account
-    #[account(
-        init_if_needed,
-        payer = payer,
-        space = Mint::LEN
-    )]
+    #[account(mut)]
     pub mint: Account<'info, Mint>,
-    // #[account(mut)]
     pub token_program: Program<'info, Token>,
-    /// CHECK: This is not dangerous because we don't read or write from this account
     #[account(mut)]
-    pub metadata: UncheckedAccount<'info>,
-    /// CHECK: This is not dangerous because we don't read or write from this account
+    pub metadata: AccountInfo<'info>,
```

Comment:
Please use UncheckedAccount instead of AccountInfo - it is the same but enforces the /// CHECK comment requirement

But if possible, use a different type. An example is token_account, which can be (I think) Account<'info, TokenAccount>.